### PR TITLE
Implement ticks

### PR DESCRIPTION
**Pull request by orestisf1993**
Adds the send_tick message and the tick event.

For info see:
- https://github.com/i3/i3/pull/3005
- https://i3wm.org/docs/ipc.html#_tick_reply
- https://i3wm.org/docs/ipc.html#_tick_event

Subscription will fail without https://github.com/i3/i3/pull/3271

### CODE REVIEW

**Comment by acrisci** [comment]
Thanks for implementing this. It could use a test and maybe better handling of the subscription failing.

**Comment by orestisf1993** [comment]
Up.

File: test/test_ticks.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        self.events.append(e)
+        if len(self.events) == 3:
+            i3.main_quit()
+
+    def test_tick_event(self, i3):
+        i3.on('tick', self.on_tick)
+        i3.event_socket_setup()
+        i3.send_tick()
+        i3.send_tick('hello world')
+        while not i3.event_socket_poll():
```
Comment (acrisci) [inline test/test_ticks.py]: This hangs forever if ticks are not received.

File: test/test_ticks.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+class TestTicks(IpcTest):
+    events = []
+
+    def on_tick(self, i3, e):
+        self.events.append(e)
+        if len(self.events) == 3:
+            i3.main_quit()
+
+    def test_tick_event(self, i3):
```
Comment (acrisci) [inline test/test_ticks.py]: Can you skip if the version is too old?

File: test/test_ticks.py (hunk 1 above)
Comment (orestisf1993) [reply test/test_ticks.py]: I guess there is no other way than copying the whole `.main()`?

Comment (acrisci) [reply test/test_ticks.py]: I put a `timeout` param in `main` for just this purpose. Maybe that should be pushed up into the `event_socket_poll` function?

Comment (orestisf1993) [reply test/test_ticks.py]: Not a bad idea, if optional.

**Comment by acrisci** [comment]
:+1:

**Pull request merged**